docs/volume-statuses.md

```diff
@@ -0,0 +1,58 @@
+# Volume statuses
+
+Last updated: 01.02.22
+
+## Abstract
+
+Every object can be created and have many states during its life.
+Transfer from one state to another should be described by final state machine (FSM).
+This document describes implemented volume statuses which should provide CSI workflow.
+
+## Background
+
+CSI specification doesn't make restrictions for volume statuses.
+It only suggests the lifecycle of a volume, but implementation of the all RPC-es has to be mandatory.
+
+## Implementation details
+
+It's been proposed to use special field of volume structure where volume status can be stored.
+That field is called "CSIStatus":
+```go
+type Volume struct {
+	Id                   string
+	Owners               []string
+	...
+	CSIStatus            string 
+	...
+```
+Now it has the next volume statuses:
+* Creating;
+* Created;
+* VolumeReady;
+* Published;
+* Removing;
+* Removed;
+* Failed;
+* Resizing;
+* Resized.
+
+FSM of volume statuses is shown below:
```

Comment:
Is RESIZED a termination state? I think we have the option to move to CREATED or VOLUME_READY from this one